pkg/timescale/write.go

```diff
@@ -112,35 +147,41 @@ func contains(s []string, e string) bool {
 }
 
 func insertRowsTimescale(rowsArray []TimescaleRows, config *config.OutputTimescale) error {
-	var c, parseErr = pgx.ParseConnectionString(config.Connection)
-	if parseErr != nil {
-		return parseErr
-	}
 
-	conn, connErr := pgx.Connect(c)
+	ctx := context.Background()
+
+	conn, connErr := timescalePools[config.Connection].Acquire(ctx)
+
 	if connErr != nil {
 		return connErr
 	}
-	defer conn.Close()
 
-	tx, beginErr := conn.Begin()
+	defer conn.Release()
+
+	tx, beginErr := conn.Begin(ctx)
 	if beginErr != nil {
 		return beginErr
 	}
-	defer tx.Rollback() //nolint: errcheck
```

Comment:
Why did you remove this here and moved the Rollback into the error handling? It was working well before... 

Maybe you were worried about calling Rollbackj on a Comitted transaction? That's ok to do and does not do any harm. See here: https://pkg.go.dev/github.com/jackc/pgx/v4#hdr-Transactions